tools/smithy-rs-sync/src/fs.rs

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use super::here;
+use anyhow::Context;
+use std::error::Error;
+use std::fmt::Display;
+use std::path::{Path, PathBuf};
+
+pub static HANDWRITTEN_DOTFILE: &str = ".handwritten";
+
+pub fn delete_all_generated_files_and_folders(directory: &Path) -> anyhow::Result<()> {
+    fn callback(path: &Path) -> Result<(), std::io::Error> {
```

Comment:
could rename these to be more descriptive, eg. `rm_path`

Reply:
I deleted the callback style functions so this is a moot point